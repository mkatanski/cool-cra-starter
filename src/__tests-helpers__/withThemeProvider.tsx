import React from "react";
import { ThemeContext } from "components/theme/ThemeProvider";
import { StyledTheme } from "styles/_types";
import { ThemeName } from "config/themes";
import { ThemeProvider } from "styled-components";
import { palette } from "styles/palette";

const TestTheme: StyledTheme = {
  container: {
    background: "container.background",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  button: {
    text: {
      default: "palette.holly",
      hover: "palette.outerSpace",
    },
    background: {
      default: "palette.azalea",
      hover: "palette.charm",
    },
  },

  input: {
    background: {
      default: "palette.azalea",
      hover: "palette.azalea",
    },
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  table: {
    background: "palette.beautyBush",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  tableHeader: {
    background: "palette.beautyBush",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  tableRowEven: {
    background: "palette.wewak",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  tableRowOdd: {
    background: "palette.beautyBush",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },
};

export const withThemeProvider = (Component: React.ReactNode) => (
  <ThemeContext.Provider
    value={{
      themeMode: ThemeName.Light,
      changeTheme: jest.fn(),
    }}
  >
    <ThemeProvider
      theme={{
        elements: TestTheme,
        palette: palette,
      }}
    >
      {Component}
    </ThemeProvider>
  </ThemeContext.Provider>
);
