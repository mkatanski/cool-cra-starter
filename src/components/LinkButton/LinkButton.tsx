import styled from "styled-components";

export type TabProps = {
  alignRight?: boolean;
};

export const LinkButton = styled.a<TabProps>`
  background: ${({ theme }) => theme.elements.button.background.default};
  color: ${({ theme }) => theme.elements.button.text.default};
  cursor: pointer;
  padding: 10px;

  display: inline-block;
  text-align: center;

  &:hover {
    background: ${({ theme }) => theme.elements.button.background.hover};
    color: ${({ theme }) => theme.elements.button.text.hover};
  }
`;
