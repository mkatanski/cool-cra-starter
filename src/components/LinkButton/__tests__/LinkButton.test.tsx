import React from "react";
import TestRenderer from "react-test-renderer";

import { LinkButton } from "../LinkButton";
import { withThemeProvider } from "__tests-helpers__/withThemeProvider";

describe("components", () => {
  describe("LinkButton", () => {
    it("should render LinkButton component", () => {
      const tree = TestRenderer.create(withThemeProvider(<LinkButton />));

      expect(tree).toMatchSnapshot();
    });
  });
});
