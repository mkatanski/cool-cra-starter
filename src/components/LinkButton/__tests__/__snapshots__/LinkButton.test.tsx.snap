// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components LinkButton should render LinkButton component 1`] = `
.c0 {
  background: palette.azalea;
  color: palette.holly;
  cursor: pointer;
  padding: 10px;
  display: inline-block;
  text-align: center;
}

.c0:hover {
  background: palette.charm;
  color: palette.outerSpace;
}

<a
  className="c0"
/>
`;
